builtin/branch.c

```diff
@@ -263,9 +265,25 @@ static int delete_branches(int argc, const char **argv, int force, int kinds,
 					| RESOLVE_REF_ALLOW_BAD_NAME,
 					&oid, &flags);
 		if (!target) {
-			error(remote_branch
-			      ? _("remote-tracking branch '%s' not found.")
-			      : _("branch '%s' not found."), bname.buf);
+			char* MISSING_REMOTE_REF_ERROR_MSG = "remote-tracking branch '%s' not found.";
+			char* MISSING_BRANCH_ERROR_MSG = "branch '%s' not found.";
+			char* MISSING_BRANCH_HINT_MSG = "branch '%s' not found.\n"
+											"Did you forget --remote?";
+
+			if (remote_branch) {
+				error(_(MISSING_REMOTE_REF_ERROR_MSG), bname.buf);
+			} else {
+				char* virtual_name = mkpathdup(FMT_REMOTES, bname.buf);
```

Comment:
I would wager a bet that this `virtual_name` leaks and is the reason for the failures in `linux-leaks`, e.g. [here](https://github.com/git/git/actions/runs/4481323437/jobs/7877870324?pr=1476#step:4:1833).

@ctmbl could you try whether adding a `FREE_AND_NULL(virtual_name)` after the `resolve_refdup()` call fixes it?

Reply:
Oh yes ofc I should have think about it! Tks for the review, I'll add it right away.